File: Solutions/1800/1800.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/10/7 10:50
# File: 1800.py
# Desc: 

class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        temp = ret = nums[0]
        for i in range(1,len(nums)):
            if nums[i] > nums[i-1]:
                temp += nums[i]
            else:
                temp = nums[i]
            ret = max(temp, ret)
        return ret

File: Solutions/1800/1800.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1800.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-10-7 18:42:36
***************************************************************************************************/


int maxAscendingSum(int* nums, int numsSize){
    int temp=nums[0], ret=nums[0];
    for (int i=1; i<numsSize; i++) {
        if (nums[i] > nums[i-1]) temp += nums[i];
        else temp = nums[i];
        ret = fmax(temp, ret);
    }
    return ret;
}
